source/common/http/utility.cc

```diff
@@ -319,5 +319,29 @@ void Utility::appendToHeader(HeaderString& header, const std::string& data) {
   header.append(data.c_str(), data.size());
 }
 
+void Utility::extractHostPathFromUri(const std::string& uri, std::string& host, std::string& path) {
+  /**
+   *  URI RFC: https://www.ietf.org/rfc/rfc2396.txt
+   *
+   *  Example:
+   *  uri  = "https://example.com/certs"
+   *  pos  :          ^
+   *  pos1 :                     ^
+   *  host = "example.com"
+   *  path = "/certs"
+   */
+  auto pos = uri.find("://");
+  pos = pos == std::string::npos ? 0 : pos + 3; // Start position of host
+  auto pos1 = uri.find("/", pos);
+  if (pos1 == std::string::npos) {
+    // If uri doesn't have "/", the whole string is treated as host.
+    host = uri.substr(pos);
+    path = "/";
+  } else {
+    host = uri.substr(pos, pos1 - pos);
+    path = "/" + uri.substr(pos1 + 1);
+  }
```

Comment:
Just a thought; would a regex possibly be clearer?

Reply:
I think regex is much slower than string.find()

Reply:
Maybe; but if you precompile the regex it could be fairly reasonable, see the ongoing discussion in https://github.com/envoyproxy/envoy/pull/3269#discussion_r189031340.